core/src/services/fs/backend.rs

```diff
@@ -566,10 +566,15 @@ impl Accessor for FsBackend {
             (p, None)
         };
 
-        let f = std::fs::OpenOptions::new()
-            .create(true)
-            .truncate(true)
-            .write(true)
+        let mut open_option = std::fs::OpenOptions::new();
+        open_option.create(true).write(true);
+        if args.append() {
```

Comment:
We should handle append carefully since the file could already been exist.

Reply:
What do you mean? 🤔 Is there any problem?

Reply:
Think about this case: If `atmoic_write` has been enabled, user trying to `append` a file. Current logic will wirte a new file in tmp_path and rename to target_path which will overwrite the existing file.

Reply:
https://github.com/apache/incubator-opendal/pull/3023#issuecomment-1711400298

That's exactly what I'm talking about.